test/workers/backend_delete_stats_worker_test.rb

```diff
@@ -0,0 +1,40 @@
+# frozen_string_literal: true
+
+require 'test_helper'
+
+class BackendDeleteStatsWorkerTest < ActiveSupport::TestCase
+  def setup
+    @service = FactoryBot.create(:simple_service)
+    @applications = FactoryBot.create_list(:cinstance, 3)
+    applications.each { |cinstance| DeletedObject.create(owner: service, object: cinstance) }
+    @metrics = FactoryBot.create_list(:metric, 3)
+    metrics.each { |metric| DeletedObject.create(owner: service, object: metric) }
+
+    @event = Services::ServiceDeletedEvent.create(service)
+    Rails.application.config.event_store.publish_event(event)
+  end
+
+  attr_reader :service, :applications, :metrics, :event
+
+  test 'perform' do
+    Timecop.freeze do
+      ThreeScale::Core::Service.expects(:delete_stats).with do |service_id, delete_job|
+        service_id == service.id && assert_delete_job_params(delete_job)
+      end
+
+      Sidekiq::Testing.inline! { BackendDeleteStatsWorker.perform_async(event.event_id) }
+    end
+  end
+
+  private
+
+  def assert_delete_job_params(delete_job)
+    return false unless delete_job.is_a? Hash
+    return false if (delete_job[:applications] || []).sort != applications.map(&:id).sort
+    return false if (delete_job[:metrics] || []).sort != metrics.map(&:id).sort
+    return false if delete_job[:users] != []
+    return false if delete_job[:from] != service.created_at.utc.to_i
+    return false if delete_job[:to] != Time.now.utc.to_i
+    true
```

Comment:
It can be refactored to

```ruby
[
  !delete_job.is_a? Hash,
  (delete_job[:applications] || []).sort != applications.map(&:id).sort,
  (delete_job[:metrics] || []).sort != metrics.map(&:id).sort,
  delete_job[:users] != [],
  delete_job[:from] != service.created_at.utc.to_i,
  delete_job[:to] != Time.now.utc.to_i
].all?
```

or in case you don't want/cannot resolve the whole array in advance:

```ruby
[
  -> { !delete_job.is_a? Hash } },
  -> { (delete_job[:applications] || []).sort != applications.map(&:id).sort },
  -> { (delete_job[:metrics] || []).sort != metrics.map(&:id).sort },
  -> { delete_job[:users] != [] },
  -> { delete_job[:from] != service.created_at.utc.to_i },
  -> { delete_job[:to] != Time.now.utc.to_i }
].all?(&:call)
```

Reply:
Good idea, thanks 👍 💪 I've done it like this:
```ruby
def assert_delete_job_params(delete_job)
    [
      -> { delete_job.is_a?(Hash) },
      -> { (delete_job[:applications] || []).sort == applications.map(&:id).sort },
      -> { (delete_job[:metrics] || []).sort == metrics.map(&:id).sort },
      -> { delete_job[:users] == [] },
      -> { delete_job[:from] == service.created_at.utc.to_i },
      -> { delete_job[:to] == Time.now.utc.to_i }
    ].all?(&:call)
  end
```

I will push when the tests for Oracle in CircleCI finish 😄

Reply:
Ok, time to push 👀 
![image](https://user-images.githubusercontent.com/11318903/53954979-0b682900-40d8-11e9-9418-3fa789f13a82.png)